packages/components/managed-data-table/src/data-table-settings/column-manager/column-manager.js

```diff
@@ -0,0 +1,246 @@
+                                    </Spacings.Inline>
+                                  ),
+                                }}
+                              />
+                            )}
+                            <DroppablePanel
+                              droppableId={HIDDEN_COLUMNS_PANEL}
+                              data-testid={HIDDEN_COLUMNS_PANEL}
+                              noColumnsText={formatMessage(
+                                messages.noHiddenColumnsToShow
+                              )}
+                              columns={hiddenColumns}
+                              isSearchable={props.areHiddenColumnsSearchable}
+                            />
+                          </Spacings.Stack>
+                        </DroppableContainer>
+                        <DroppableContainer isDragging={isDragging}>
+                          <Spacings.Stack>
+                            <Spacings.Inline scale="xs" alignItems="center">
+                              <EyeIcon size="medium" />
+                              <FieldLabel
+                                title={formatMessage(messages.visibleColumns)}
+                              />
+                            </Spacings.Inline>
+                            <DroppablePanel
+                              droppableId={SELECTED_COLUMNS_PANEL}
+                              data-testid={SELECTED_COLUMNS_PANEL}
+                              noColumnsText={formatMessage(
+                                messages.noSelectedColumnsToShow
+                              )}
+                              columns={props.selectedColumns}
+                              onRemove={props.onUpdateColumns}
+                            />
+                          </Spacings.Stack>
+                        </DroppableContainer>
+                      </Spacings.Inline>
+                    </DragDropContext>
+                  </div>
+                </div>
+              </Spacings.Inset>
+              {(props.secondaryButton || props.primaryButton) && (
+                <Spacings.Inline justifyContent="flex-end">
+                  {props.secondaryButton}
+                  {props.primaryButton}
+                </Spacings.Inline>
+              )}
+            </Spacings.Stack>
+          </Spacings.Stack>
+        </Container>
+      )}
+    </CollapsibleMotion>
+  );
+};
+
+ColumnManager.displayName = 'ColumnManager';
+
+ColumnManager.propTypes = {
+  availableColumns: PropTypes.arrayOf(
+    PropTypes.shape({
+      key: PropTypes.string.isRequired,
+      label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
+    })
+  ).isRequired,
+  selectedColumns: PropTypes.arrayOf(
+    PropTypes.shape({
+      key: PropTypes.string.isRequired,
+      label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
+    })
+  ).isRequired,
+  onUpdateColumns: PropTypes.func.isRequired,
+
+  areHiddenColumnsSearchable: PropTypes.bool,
```

Comment:
Could this be flipped? UIKit has a lot of `disable` things. So opt-out instead of opt-in. `disableHiddenColumnSearch` which defaults to `true` maybe?